File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        count = 0;
        for i in range(len(tickets)):
            if(i<=k):
                count+= min(tickets[i],tickets[k])
            else:
                count+= min(tickets[i],tickets[k]-1)
        return count

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        int count = 0;
        int n = tickets.size();
        while(tickets[k]!=0){
            for(int i=0;i<n;i++){
                if(tickets[i]){
                    count++; 
                    tickets[i]--;
                }
                if(i==k && tickets[i]==0) return count;
            }
        }
        return 0;
    }
};